File: Betch-2-1/ClassWork/DataStructurs/Unit-1/Sorting/InsertionSort/InsertionSort in java/InsertionSort.java
import java.util.Scanner;
class InsertionSort{
    private static int arr[];
    static Scanner sc=new Scanner(System.in);

    InsertionSort(){
        read();
        sort();
        display();
    }

    private static void read(){
        System.out.print("enter no.of elements:");
        arr=new int[sc.nextInt()];
        System.out.print("enter elements:");
        for(int i=0;i<arr.length;i++){
            arr[i]=sc.nextInt();
        }
    }

    private static void sort(){
        for(int i=0;i<arr.length;i++){
            int j=i-1;
            int temp=arr[i];
            while(j>=0 && temp<arr[j]){
                arr[j+1]=arr[j];
                j-=1;
            }arr[j+1]=temp;
        }
    }
    
    private static void display(){
        System.out.print("sorted array:");
        for(int i=0;i<arr.length;i++){
            System.out.print(arr[i]+" ");
        }
    }
}

class Main{
    public static void main(String[] args) {
        InsertionSort is=new InsertionSort();
    }
}

File: B-tech-2-1/ClassWork/DataStructurs/Unit-1/Sorting/InsertionSort/InsertionSort in py/InsertionSort.py
class InsertionSort:
    def __init__(self,arr):
        self.arr=arr
        InsertionSort.sort(self)
        InsertionSort.display(self)
    @staticmethod
    def sort(self):
        for i in range(1,len(arr)):
            temp=self.arr[i]
            j = i-1
            while j>=0 and temp<self.arr[j] :
                self.arr[j+1]=self.arr[j]
                j-=1
            self.arr[j+1]=temp
    @staticmethod
    def display(self):
        print(self.arr)

arr=list(map(int,input("enter elements:").split()))
obj=InsertionSort(arr)
